TestCrossNamespace  failure

```
TestCrossNamespace/cross_namespace_access,_the_new_namespace_is_not_managed_by_Kmesh_**AGAIN**
    testing.go:1577: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
2025-04-29T08:39:47.652085Z	info	tf	=== DONE (failed):  Test: '_home_runner_work_kmesh_kmesh_test_e2e[TestCrossNamespace] (34.949151713s)' ===
--- FAIL: TestCrossNamespace (34.95s)
    --- PASS: TestCrossNamespace/cross_namespace_access,_the_new_namespace_is_not_managed_by_Kmesh (0.02s)
    --- PASS: TestCrossNamespace/cross_namespace_access,_the_new_namespace_is_managed_by_Kmesh (0.03s)
    --- FAIL: TestCrossNamespace/cross_namespace_access,_the_new_namespace_is_not_managed_by_Kmesh_**AGAIN** (30.01s
```

This is my first seeing this failure

--- a/pkg/controller/manage/manage_controller.go
+++ b/pkg/controller/manage/manage_controller.go
@@ -159,7 +159,7 @@ func (c *KmeshManageController) handlePodAdd(obj interface{}) {
 	c.enableKmeshManage(newPod)
 }
 
-func (c *KmeshManageController) handlePodUpdate(_, newObj interface{}) {
+func (c *KmeshManageController) handlePodUpdate(oldObj, newObj interface{}) {
 	pod, ok := newObj.(*corev1.Pod)
 	if !ok {
 		log.Errorf("expected *corev1.Pod but got %T", newObj)
@@ -169,6 +169,20 @@ func (c *KmeshManageController) handlePodUpdate(_, newObj interface{}) {
 		log.Debugf("pod %s/%s is being deleted, skip remanage", pod.Namespace, pod.Name)
 		return
 	}
+
+	oldPod, ok := oldObj.(*corev1.Pod)
+	if !ok {
+		log.Errorf("expected *corev1.Pod but got %T", oldObj)
+		return
+	}
+
+	// We don't need to remanage the pod if the annotation of KmeshRedirectionAnnotation is changed.
+	// Ref: https://github.com/kmesh-net/kmesh/issues/1357
+	if oldPod.Annotations[constants.KmeshRedirectionAnnotation] != pod.Annotations[constants.KmeshRedirectionAnnotation] {
+		log.Debugf("pod %s/%s annotation of KmeshRedirectionAnnotation is changed, skip remanage", pod.Namespace, pod.Name)
+		return
+	}
+
 	c.handlePodAdd(newObj)
 }
 

